src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java

```diff
@@ -0,0 +1,299 @@
+    {
+        return buildingLevel;
+    }
+
+    /**
+     * Get the max level of the building.
+     *
+     * @return AbstractBuilding max level.
+     */
+    public int getBuildingMaxLevel()
+    {
+        return buildingMaxLevel;
+    }
+
+    /**
+     * Checks if this building is at its max level.
+     *
+     * @return true if the building is at its max level.
+     */
+    public boolean isBuildingMaxLevel()
+    {
+        return buildingLevel >= buildingMaxLevel;
+    }
+
+    /**
+     * Get the current work order level.
+     *
+     * @return 0 if none, othewise the current level worked on
+     */
+    public int getCurrentWorkOrderLevel()
+    {
+        return workOrderLevel;
+    }
+
+    /**
+     * Get the current work order level.
+     *
+     * @return 0 if none, othewise the current level worked on
+     */
+    public boolean hasWorkOrder()
+    {
+        return workOrderLevel != NO_WORK_ORDER;
+    }
+
+    public boolean isBuilding()
+    {
+        return workOrderLevel != NO_WORK_ORDER && workOrderLevel > buildingLevel;
+    }
+
+    public boolean isRepairing()
+    {
+        return workOrderLevel != NO_WORK_ORDER && workOrderLevel == buildingLevel;
+    }
+
+    /**
+     * Open the associated BlockOut window for this building.
+     */
+    public void openGui()
+    {
+        @Nullable final Window window = getWindow();
+        if (window != null)
+        {
+            window.open();
+        }
+    }
+
+    /**
+     * Will return the window if this building has an associated BlockOut window.
+     *
+     * @return BlockOut window.
+     */
+    @Nullable
+    public Window getWindow()
+    {
+        return null;
+    }
+
+    /**
+     * Read this view from a {@link ByteBuf}.
+     *
+     * @param buf The buffer to read this view from.
+     */
+    public void deserialize(@NotNull final ByteBuf buf)
+    {
+        buildingLevel = buf.readInt();
+        buildingMaxLevel = buf.readInt();
+        buildingDmPrio = buf.readInt();
+        workOrderLevel = buf.readInt();
+
+        loadRequestSystemFromNBT(ByteBufUtils.readTag(buf));
+    }
+
+    private void loadRequestSystemFromNBT(NBTTagCompound compound)
+    {
+        this.citizensByRequests.clear();
+        if (compound.hasKey(TAG_CITIZEN_BY_REQUEST))
+        {
+            NBTTagList citizensByRequestList = compound.getTagList(TAG_CITIZEN_BY_REQUEST, Constants.NBT.TAG_COMPOUND);
+            NBTUtils.streamCompound(citizensByRequestList).forEach(cbrc -> processIntegerKeyTokenList(cbrc, citizensByRequests));
+        }
+
+
+        this.requestsByCitizen.clear();
+        this.citizensByRequests.keySet().forEach(citizen -> this.citizensByRequests.get(citizen).forEach(requestToken -> this.requestsByCitizen.put(requestToken, citizen)));
+    }
+
+    @SuppressWarnings(Suppression.GENERIC_WILDCARD)
+    public <R> ImmutableList<IRequest<? extends R>> getOpenRequestsOfType(@NotNull final CitizenDataView citizenData, final Class<R> requestType)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)